lib/data/wrapper.js

```diff
@@ -219,16 +230,23 @@ const data = {
 
     delete: (objectGetInfo, log, cb) => {
         const callback = cb || log.end;
-        // If objectGetInfo.key exists the md-model-version is 2 or greater.
-        // Otherwise, the objectGetInfo is just the key string.
-        const objGetInfo = (implName === 'sproxyd' || implName === 'cdmi') ?
-            objectGetInfo.key : objectGetInfo;
+        const isMdModelVersion2 = typeof(objectGetInfo) === 'string';
```

Comment:
Eventually we would better use `ObjectMD` class to centralize compat stuff in a single location instead of having to deal with this at various locations and take the risk of screwing or forgetting to handle it sometimes. But of course this is not the scope here.

Reply:
one hundred percent positive about this